flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -106,6 +112,10 @@ class DataStreamOverAggregate(
         if (overWindow.lowerBound.isUnbounded &&
           overWindow.upperBound.isCurrentRow) {
           createUnboundedAndCurrentRowProcessingTimeOverWindow(inputDS)
+        } // lowerbound is a RowType and upperbound is PRECEEDING or CURRENT ROW
+        else if (overWindow.isRows &&
+            (overWindow.upperBound.isPreceding || overWindow.upperBound.isCurrentRow)) {
```

Comment:
If the upper bound is not specified the default is `CURRENT ROW`. So this check can be safely added, IMO. Just debug the case of `OVER (ORDER BY procTime() ROWS 2 PRECEDING)`.